### PR TITLE
Drawer 

### PR DESCRIPTION
**Pull request by JoanaMMoreira**
Adds the Drawer component to Lab, including the following features:

- Left (default), right, top and bottom positioning
- Enable (default)/ disable animations
- Enable (default)/ disable `Scrim`
- Primary (default) and secondary variants

Updated examples will be added later on

**20/07 update** - I'll raise a follow up PR to replace `Scrim` with floating-ui's `FloatingFocusManager`

### CODE REVIEW

File: .changeset/fifty-buses-compete.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+"@salt-ds/lab": patch
+---
+
+Drawer
+
+- Added `Drawer` component
```
Comment (Fercas123) [inline .changeset/fifty-buses-compete.md:7]: we've been adding a code block with default usage examples to the changeset, you might need one too

**Review by Fercas123 — APPROVED** [review]
Looks good! just a small comment, probably not blocker

**Review by Fercas123 — APPROVED** [review]

File: packages/lab/stories/drawer/drawer.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+} from "@salt-ds/core";
+import { CloseIcon } from "@salt-ds/icons";
+import { Drawer, DRAWER_POSITIONS } from "@salt-ds/lab";
+import { ComponentMeta, ComponentStory } from "@storybook/react";
+import "./drawer.stories.css";
+
+export default {
+  title: "Lab/Drawer",
+  component: Drawer,
+  argTypes: {
```
Comment (joshwooding) [inline packages/lab/stories/drawer/drawer.stories.tsx:21]: Was there an issue relying on the default types?

File: packages/lab/src/drawer/Drawer.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  } = props;
+
+  const targetWindow = useWindow();
+  useComponentCssInjection({
+    testId: "salt-drawer",
+    css: drawerCss,
+    window: targetWindow,
+  });
+
+  const [showComponent, setShowComponent] = useState(false);
```
Comment (joshwooding) [inline packages/lab/src/drawer/Drawer.tsx:70]: We should talk about whether we want to unmount the drawer contents. If we don't this component gets a lot simpler.

File: packages/lab/stories/drawer/drawer.stories.tsx (hunk 2 above)
Comment (JoanaMMoreira) [reply packages/lab/stories/drawer/drawer.stories.tsx]: This was just for adding the storybook dropdown which isn't really necessary so I've removed it

File: packages/lab/src/drawer/Drawer.css [hunk 4]
```diff
@@ -0,0 +1,98 @@
+.saltDrawer {
+  --drawer-boxShadow: var(--saltDrawer-boxShadow, var(--salt-overlayable-shadow-modal));
```
Comment (libertymayc) [inline packages/lab/src/drawer/Drawer.css:2]: doesn't need to be a local token. also, the CSS api is defined here and on the attr (line 24) which is unnecessary - but, I think we aren't adding CSS API tokens to new components atm, so can be deleted and use the characteristic directly :)

Comment (libertymayc) [reply packages/lab/src/drawer/Drawer.css:2]: same for all below

File: packages/lab/src/drawer/Drawer.css [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+.saltDrawer-left.saltDrawer-exit-animation {
+  animation: var(--salt-animation-slide-out-left);
+}
+
+.saltDrawer-bottom {
+  left: 0;
+  right: 0;
+  bottom: 0;
+  top: auto;
```
Comment (libertymayc) [inline packages/lab/src/drawer/Drawer.css:89]: auto is default, could delete

Comment (libertymayc) [reply packages/lab/src/drawer/Drawer.css:89]: same for others where value is auto

File: packages/lab/src/drawer/Drawer.tsx [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+  const exitAnimation = !disableAnimations && !isOpen;
+
+  const drawer = showComponent ? (
+    <div
+      ref={ref}
+      className={clsx(withBaseName(), className, withBaseName(position), {
+        [withBaseName("enter-animation")]: enterAnimation,
+        [withBaseName("exit-animation")]: exitAnimation,
+        [withBaseName(variant)]: variant,
+      })}
```
Comment (libertymayc) [inline packages/lab/src/drawer/Drawer.tsx:95]:
```suggestion
      className={clsx(withBaseName(), withBaseName(position), {
        [withBaseName("enter-animation")]: enterAnimation,
        [withBaseName("exit-animation")]: exitAnimation,
        [withBaseName(variant)]: variant,
      }, className)}
```

File: packages/theme/css/foundations/animation.css [hunk 7]
```diff
@@ -17,10 +17,10 @@
...
   --salt-animation-slide-in-bottom: slide-in-bottom var(--salt-animation-duration) var(--salt-animation-timing-function);
 
-  --salt-animation-slide-out-top: slide-out-top var(--salt-animation-duration) var(--salt-animation-timing-function);
-  --salt-animation-slide-out-left: slide-out-left var(--salt-animation-duration) var(--salt-animation-timing-function);
-  --salt-animation-slide-out-right: slide-out-right var(--salt-animation-duration) var(--salt-animation-timing-function);
-  --salt-animation-slide-out-bottom: slide-out-bottom var(--salt-animation-duration) var(--salt-animation-timing-function);
+  --salt-animation-slide-out-top: slide-out-top var(--salt-animation-duration) var(--salt-animation-timing-function) both;
+  --salt-animation-slide-out-left: slide-out-left var(--salt-animation-duration) var(--salt-animation-timing-function) both;
+  --salt-animation-slide-out-right: slide-out-right var(--salt-animation-duration) var(--salt-animation-timing-function) both;
+  --salt-animation-slide-out-bottom: slide-out-bottom var(--salt-animation-duration) var(--salt-animation-timing-function) both;
```
Comment (libertymayc) [inline packages/theme/css/foundations/animation.css:23]: can you add these to a patch changeset for theme as value changed

File: packages/lab/src/drawer/Drawer.tsx (hunk 3 above)
Comment (libertymayc) [inline packages/lab/src/drawer/Drawer.tsx:70]: should click outside drawer/escape key close it?

Comment (JoanaMMoreira) [reply packages/lab/src/drawer/Drawer.tsx:70]: I had a look and it does seem that most other libraries adopt the same behaviour, let's discuss tomorrow on the dev call :)

File: packages/lab/src/drawer/Drawer.tsx [hunk 8]
```diff
@@ -0,0 +1,123 @@
+import { forwardRef, HTMLAttributes, useEffect, useState } from "react";
+import { clsx } from "clsx";
+
+import { Scrim, ScrimProps } from "../scrim";
```
[the remark this answers is not in the document]
Comment (JoanaMMoreira) [reply packages/lab/src/drawer/Drawer.tsx:4]: cc @bhoppers2008

File: .changeset/strong-impalas-eat.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
+---
+"@salt-ds/lab": patch
+"@salt-ds/theme": patch
+---
+
+Drawer
+
+- Added `Drawer` component
+  JoanaMMoreira marked this conversation as resolved.
```
Comment (feliciackh) [inline .changeset/strong-impalas-eat.md:9]: think this line was added by mistake

File: packages/lab/src/drawer/Drawer.tsx [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+  const exitAnimation = !disableAnimations && !isOpen;
+
+  const drawer = showComponent ? (
+    <div
+      ref={ref}
+      className={clsx(
+        withBaseName(),
+        withBaseName(position),
+        {
+          [withBaseName("enter-animation")]: enterAnimation,
```
Comment (feliciackh) [inline packages/lab/src/drawer/Drawer.tsx:95]: i think we've been using camel cases for classNames, might be a good idea to align that with other components

**Comment by bhoppers2008** [comment]
Update: @JoanaMMoreira @joshwooding to discuss what to do about Scrim.

File: .changeset/strong-impalas-eat.md (hunk 9 above)
Comment (JoanaMMoreira) [reply .changeset/strong-impalas-eat.md]: good spot 😆

**Review by joshwooding — DISMISSED** [review]

**Review by alycrys — APPROVED** [review]

File: .changeset/strong-impalas-eat.md [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+   <p>Drawer content</p>
+   <Button onClick={hide}>Close Drawer</Button>
+  </div>
+ </Drawer>
+</>
+```
+
+Theme
+
+- Added `both` fill-mode to `--salt-animation-slide-out-*` css animation vars
```
Comment (alycrys) [inline .changeset/strong-impalas-eat.md:27]: should there be two changesets?

**Review by feliciackh — APPROVED** [review]

File: .changeset/strong-impalas-eat.md (hunk 11 above)
Comment (JoanaMMoreira) [reply .changeset/strong-impalas-eat.md:27]: > should there be two changesets?

do we usually split them? I'm not sure but happy to do so if that's the case

Comment (joshwooding) [reply .changeset/strong-impalas-eat.md:27]: Yeah, this should be split.

**Review by joshwooding — APPROVED** [review]

**Review by libertymayc — APPROVED** [review]

**Pull request merged**